File: component-service/src/main/java/dev/yidafu/pan/component/service/ComponentService.java
package dev.yidafu.pan.component.service;

import dev.yidafu.pan.component.domain.dto.SaveComponentDTO;
import dev.yidafu.pan.component.domain.dto.UpdateComponentDTO;
import dev.yidafu.pan.component.domain.vo.ComponentVO;

import java.util.List;

public interface ComponentService {
    ComponentVO findById(Long id);

    List<ComponentVO> findAllByScreenId(Long screenId);

    ComponentVO updateById(Long id, UpdateComponentDTO dto);

    ComponentVO createOne(SaveComponentDTO dto);
}

File: component-service/src/main/kotlin/dev/yidafu/pan/component/service/ComponentService.kt
package dev.yidafu.pan.component.service

import dev.yidafu.pan.common.model.dto.SaveComponentDTO
import dev.yidafu.pan.common.model.dto.UpdateComponentDTO
import dev.yidafu.pan.common.model.vo.ComponentVO

interface ComponentService {
    fun findById(id: Long): ComponentVO
    fun findAllByScreenId(screenId: Long): List<ComponentVO>
    fun updateById(id: Long, dto: UpdateComponentDTO): ComponentVO

    /**
     * 创建组件
     */
    fun createOne(dto: SaveComponentDTO): ComponentVO

    /**
     * 根据 id 删除组件
     */
    fun removeById(id: Long): Boolean

    fun templateList(): List<ComponentVO>
}
